src/lib/helpers/fetchToken.js

```diff
@@ -3,17 +3,22 @@ export const fetchToken = ({ clientId, clientSecret, code, verifier, tokenEndpoi
     client_id: clientId,
     code,
     grant_type: 'authorization_code',
-    code_verifier: verifier
+    code_verifier: verifier,
+    redirectUri,
+    useUrlEncodedForm,
   };
   if (clientSecret) {
     payload.client_secret = clientSecret
   }
+
+  var body = useUrlEncodedForm ? Object.keys(payload).map(key => key + '=' + payload[key]).join('&') : JSON.stringify(payload);
+
   return fetch(tokenEndpoint, {
     headers: {
-      'Content-Type': 'application/json'
+      'Content-Type': useUrlEncodedForm ? 'application/x-www-form-urlencoded' : 'application/json'
     },
     method: 'POST',
-    body: JSON.stringify(payload)
+    body: JSON.stringify(body)
```

Comment:
This looks wrong: If `useUrlEncodedForm` is `false`, then you stringify twice, and if `useUrlEncodedForm` is `true`, then the body should not be stringified, I suppose.